game-of-life.js

```diff
@@ -0,0 +1,249 @@
+let matrix; 
+let spanMouseX = document.getElementById('mouse-x'); 
+let spanMouseY = document.getElementById('mouse-y');
+let canvas = document.getElementById('canvas');
+let context = canvas.getContext("2d");
+const cellSize = 10;
+let body = document.getElementById('body').addEventListener('click', handleMouseClick);
+let status;
+
+window.addEventListener('DOMContentLoaded', init);
+
+function init(){
+    console.log('Hello from init');
+}
+
+function handleMouseClick(event){
+    const boundingRect = canvas.getBoundingClientRect();
+    let point = calcClickCoordinates(event.clientX, event.clientY, boundingRect.left, boundingRect.top, boundingRect.right, boundingRect.bottom);
+    if(point){
+        let point2 = calcCellCoordinates(point.x , point.y);
+        if(matrix[point2.y][point2.x] === 0){
+            matrix[point2.y][point2.x] = 1;
+        }
+        else{
+            matrix[point2.y][point2.x] = 0;
+        }
+        draw();
+        spanMouseX.innerHTML = 'Column ' + point2.x;
+        spanMouseY.innerHTML = 'Row ' + point2.y;
+    }
+    else {
+        spanMouseX.innerHTML = 'Out of bounds';
+        spanMouseY.innerHTML = 'Out of bounds';
+    }
+}
+function calcCellCoordinates(canvasX, canvasY){
+    let colNum = (Math.floor((canvasX / cellSize)));
+    let rowNum = (Math.floor((canvasY / cellSize)));
+    let point2 = {x: colNum,y: rowNum};
+    return point2;
+}
+
+function calcClickCoordinates(mouseX, mouseY, canvasLeft, canvasTop, canvasRight, canvasBottom){
+    let xInBounds = false;
+    let yInBounds = false;
+    if (mouseX >= canvasLeft && mouseX <= canvasRight){
+        xInBounds = true;
+    }
+    if (mouseY >= canvasTop && mouseY <= canvasBottom){
+        yInBounds = true;
+    }
+    if (xInBounds && yInBounds) {
+        let clickX = mouseX - canvasLeft;
+        let clickY = mouseY - canvasTop;
+        let point = {x: clickX, y: clickY};
+        return point;
+    }
+    else {
+        return null;
+    }
+}
+function draw(){
+    for (let i = 0; i < matrix.length; i++){
+        for (let j = 0; j < matrix[0].length; j++){
+            if(matrix[i][j] === 1){
+                context.fillStyle = 'black';
+                context.fillRect(j*cellSize + 1, i*cellSize + 1, cellSize - 2, cellSize - 2);
+            }
+            else{
+                context.fillStyle = 'white';
+                context.fillRect(j *cellSize + 1, i *cellSize + 1, cellSize - 2, cellSize - 2); 
+            }
+        }
+    }
+}
+function simulate(){
+    console.log('INITIALIZE');
+    for (let i = 0; i < matrix.length; i++){
+        for (let j = 0; j < matrix[0].length; j++){
+        
+            let reindexJ = j;
+            let reindexI = i;
+
+            reindexJ = reindexJ + 1;
+            reindexI = reindex(reindexI);
+            reindexJ = reindex(reindexJ);
+            let rightNeighbourCount = matrix[reindexI][reindexJ];
+            reindexJ = reindexJ - 1;
```

Comment:
Looks like a repeating function that could be moved into a helper function. `checkNeighbour(x,y,xOffset,yOffset)`